sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBSuite.scala

```diff
@@ -1736,6 +1737,84 @@ class RocksDBSuite extends AlsoTestWithChangelogCheckpointingEnabled with Shared
     }
   }
 
+  testWithChangelogCheckpointingEnabled("load the same version of pending snapshot uploading") {
```

Comment:
should we add comments within the thread to explain what we think should happen ?

Reply:
+1 Shall we describe the scenario and the expected behavior?